### PR TITLE
storage: not using threads on read operations

### PR DESCRIPTION
**Pull request by Volham22**
# What has changed ?

This PR remove usage of separate threads when performing a reading operations. **I'm not 100% sure this would never block as get operations can update database sometimes**

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: parsec/core/fs/storage/workspace_storage.py [hunk 1]
```diff
@@ -183,7 +183,10 @@ async def run(
 
     @manage_operational_error
     async def clear_memory_cache(self, flush: bool = True) -> None:
-        return await trio.to_thread.run_sync(self.sync_instance.clear_memory_cache, flush)
+        if flush:
+            return await trio.to_thread.run_sync(self.sync_instance.clear_memory_cache, flush)
+        else:
+            return self.sync_instance.clear_memory_cache(flush)
```
Comment (FirelightFlagboy) [inline parsec/core/fs/storage/workspace_storage.py:189]:
```suggestion
            return await trio.to_thread.run_sync(self.sync_instance.clear_memory_cache, True)
        else:
            return self.sync_instance.clear_memory_cache(False)
```

Could be more explicit :thinking:

File: parsec/core/fs/storage/user_storage.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ async def run(cls, data_base_dir: Path, device: LocalDevice) -> AsyncIterator["U
 
         # Populate the cache with the user manifest to be able to
         # access it synchronously at all time
-        await trio.to_thread.run_sync(self.sync_instance.load_user_manifest)
+        self.sync_instance.load_user_manifest()
```
Comment (FirelightFlagboy) [inline parsec/core/fs/storage/user_storage.py:44]: We do a `set_manifest` if the user manifest is not present in the DB

File: parsec/core/fs/storage/workspace_storage.py [hunk 3]
```diff
@@ -230,13 +233,13 @@ async def clear_clean_block(self, block_id: BlockID) -> None:
 
     @manage_operational_error
     async def get_dirty_block(self, block_id: BlockID) -> bytes:
-        return await trio.to_thread.run_sync(self.sync_instance.get_dirty_block, block_id)
+        return self.sync_instance.get_dirty_block(block_id)
```
Comment (FirelightFlagboy) [inline parsec/core/fs/storage/workspace_storage.py:236]: `get_dirty_block` update the database

File: parsec/core/fs/storage/workspace_storage.py [hunk 4]
```diff
@@ -230,13 +233,13 @@ async def clear_clean_block(self, block_id: BlockID) -> None:
...
 
     # Chunk interface
 
     @manage_operational_error
     async def get_chunk(self, chunk_id: ChunkID) -> bytes:
-        return await trio.to_thread.run_sync(self.sync_instance.get_chunk, chunk_id)
+        return self.sync_instance.get_chunk(chunk_id)
```
Comment (FirelightFlagboy) [inline parsec/core/fs/storage/workspace_storage.py:242]: `get_chunk` update the database

File: parsec/core/fs/storage/workspace_storage.py [hunk 5]
```diff
@@ -261,7 +264,7 @@ async def set_prevent_sync_pattern(self, pattern: Regex) -> None:
         This operation is idempotent,
         i.e it does not reset the `fully_applied` flag if the pattern hasn't changed.
         """
-        return await trio.to_thread.run_sync(self.sync_instance.set_prevent_sync_pattern, pattern)
+        return self.sync_instance.set_prevent_sync_pattern(pattern)
```
Comment (FirelightFlagboy) [inline parsec/core/fs/storage/workspace_storage.py:267]: `set_prevent_sync_pattern` update the database

File: parsec/core/fs/storage/workspace_storage.py [hunk 6]
```diff
@@ -271,15 +274,13 @@ async def mark_prevent_sync_pattern_fully_applied(self, pattern: Regex) -> None:
         workspace are compliant with the new pattern. The applied pattern is provided
         as an argument in order to avoid concurrency issues.
         """
-        return await trio.to_thread.run_sync(
-            self.sync_instance.mark_prevent_sync_pattern_fully_applied, pattern
-        )
+        return self.sync_instance.mark_prevent_sync_pattern_fully_applied(pattern)
```
Comment (FirelightFlagboy) [inline parsec/core/fs/storage/workspace_storage.py:277]: `mark_prevent_sync_pattern_fully_applied` update the database

File: parsec/core/fs/storage/workspace_storage.py [hunk 7]
```diff
@@ -385,7 +384,7 @@ def workspace_id(self) -> EntryID:
         return self.sync_instance.workspace_id
 
     async def get_chunk(self, chunk_id: ChunkID) -> bytes:
-        return await trio.to_thread.run_sync(self.sync_instance.get_chunk, chunk_id)
+        return self.sync_instance.get_chunk(chunk_id)
```
Comment (FirelightFlagboy) [inline parsec/core/fs/storage/workspace_storage.py:387]: `get_chunk` update the database

File: parsec/core/fs/storage/workspace_storage.py [hunk 8]
```diff
@@ -455,15 +454,15 @@ async def clear_clean_block(self, block_id: BlockID) -> None:
         return await trio.to_thread.run_sync(self.sync_instance.clear_clean_block, block_id)
 
     async def get_dirty_block(self, block_id: BlockID) -> bytes:
-        return await trio.to_thread.run_sync(self.sync_instance.get_dirty_block, block_id)
+        return self.sync_instance.get_dirty_block(block_id)
```
Comment (FirelightFlagboy) [inline parsec/core/fs/storage/workspace_storage.py:457]: `get_dirty_block` update the database

**Comment by Volham22** [comment]
Oh mb, I thought get like operations were just read to the database. Anyway I did change according to your comments and now copying the latest source code release of `parsec` takes roughly 30s.
```
corentinDesktop /tmp » time cp -r parsec-cloud-2.15.0 ~/Parsec/alice_workspace
cp -r parsec-cloud-2.15.0 ~/Parsec/alice_workspace  0,02s user 0,23s system 0% cpu 30,839 total
corentinDesktop /tmp » time cp -r parsec-cloud-2.15.0 ~/Parsec/alice_workspace
cp -r parsec-cloud-2.15.0 ~/Parsec/alice_workspace  0,04s user 0,27s system 0% cpu 33,530 total
corentinDesktop /tmp » du -sh parsec-cloud-2.15.0
26M	parsec-cloud-2.15.0
```
In comparison the same operation on master takes about 30s too on my machine.

**Review by FirelightFlagboy — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Comment by Volham22** [comment]
Should we close this PR @FirelightFlagboy ? We know that it's not going to work.

**Comment by FirelightFlagboy** [comment]
Not for the moment but soon tm

**Comment by TimeEngineer** [comment]
@Volham22 Need to rebase

**Comment by FirelightFlagboy** [comment]
@TimeEngineer not need this PR is to be merged

**Comment by TimeEngineer** [comment]
@FirelightFlagboy Can be closed ?

**Comment by FirelightFlagboy** [comment]
Yes I'm closing right now

**Pull request closed without merging**